webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+    }
+
+    descriptionSite2 = controller.getWebSite(id);
+    controller.setSessionSite(descriptionSite2);
+    request.setAttribute("Site", descriptionSite2);
+  }
+
+  private String addNewDesignedSite(final WebSiteSessionController controller, final HttpRequest request)
+      throws WebSitesException {
+    // Filled at first access then null
+    String nomSite = request.getParameter(WebSitesRequestRouter.SITE_NAME_PARAM);
+    // Filled at first access then null
+    String description = request.getParameter(WebSitesRequestRouter.DESCRIPTION_PARAM);
+    // Filled at first access then null
+    String nomPage = request.getParameter(WebSitesRequestRouter.PAGE_NAME_PARAM);
+    String tempPopup = request.getParameter(WebSitesRequestRouter.POPUP_PARAM);
+    // Filled at first creation then null
+    String listeIcones = request.getParameter(WebSitesRequestRouter.ICON_LIST_PARAM);
+    // = en cas de new Site ou de classifySite
+    String listeTopics = request.getParameter(WebSitesRequestRouter.TOPIC_LIST_PARAM);
+    // Retrieve positions
+    String positions = request.getParameter("Positions");
+
+    int popup = 0;
+    if ((tempPopup != null) && (tempPopup.length() > 0)) {
+      popup = 1;
+    }
+
+    ArrayList<String> listIcons = new ArrayList<>();
+    if (listIcons != null) {
+      for (String icon : listeIcones.split(",")) {
+        if (!icon.isEmpty()) {
+          listIcons.add(icon);
+        }
+      }
+    }
+
+          /* recuperation de l'id */
+    String id = controller.getNextId();
+
+          /* Creer le repertoire id */
+    controller.createFolder(controller.getWebSitePathById(id));
+
+    // Persist siteDetail inside database type 0 = site cree
+    SiteDetail descriptionSite =
+        new SiteDetail(id, controller.getComponentId(), nomSite, description, nomPage, 0, null, null, 0,
+            popup);
+    descriptionSite.setPositions(positions);
+
+    String pubId = controller.createWebSite(descriptionSite);
+    descriptionSite = controller.getWebSite(id);
+    controller.setSessionSite(descriptionSite);
+
+    if (listIcons.size() > 0) {
+      controller.associateIcons(id, listIcons);
+    }
+
+    String currentPath = controller.getWebSitePathById(id);
+
+          /* ajout de la page principale */
+    String code = " ";
+
+          /* Creer la page principale */
+    controller.createFile(currentPath, nomPage, code);
+
+          /* publications : classer le site dans les themes cochés */
+    ArrayList<String> arrayToClassify = new ArrayList<String>();
+    boolean publish = false;
+    for (String idTopic : listeTopics.split(",")) {
+      if (!idTopic.isEmpty()) {
+        // ajout de la publication dans le theme
+        controller.addPublicationToFolder(pubId, idTopic);
+        publish = true;
+      }
+    }
+
+    if (publish) {
+      arrayToClassify.add(id);
+      controller.publish(arrayToClassify); // set etat du site a 1
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)